AuthoringTests.md

```diff
@@ -2,39 +2,103 @@
-   1. Run the `GVFS.FunctionalTests` project.  Even better, set it as the default project and hit F5.
-   2. Use the command line runner.  After building, execute `Scripts\RunFunctionalTests.bat`
-   3. If you want to use Visual Studio's Test Explorer, you need to install the NUnit 3 Test Adapter in VS | Tools | Extensions and Updates.
+`GVFS.FunctionalTests` is a .NET Core project and contains all cross-platform functional tests.  `GVFS.FunctionalTests.Windows`, contains functional tests that require Windows. Additionally, `GVFS.FunctionalTests.Windows` includes all the `GVFS.FunctionalTests` allowing it to run both cross-platform and Windows-specific functional tests.
 
-Running the `GVFS.FunctionalTests` project is probably the most convenient for developers.  `RunFunctionalTests.bat` will be used on the build machines.
+#### Other functional test projects
+
+*GVFS.NativeTests*
+
+`GVFS.NativeTests` contains tests written in C++ that use the Windows API directly.  These tests are called from the managed tests (see above) using PInvoke.
+
+*GVFS.FunctionalTests.LockHolder*
+
+The `LockHolder` is a small program that allows the functional tests to request and release the `GVFSLock`.  `LockHolder` is useful for simulating different timing/race conditions.
+
+## Running the Functional Tests
+
+The functional tests are built on NUnit 3, which is available as a set of NuGet packages.
+
+### Windows
+
+1. Build VFS for Git:
+    
+    **Option 1:** Open GVFS.sln in Visual Studio and build everything.
+    
+    **Option 2:** Run `Scripts\BuildGVFSForWindows.bat` from the command line
+
+2. Run the VFS4G installer that was built in step 2.  This will ensure that ProjFS is properly installed/enabled on your machine, and that VFS4G will be able to find the correct version of the pre/post-command hooks. The installer will be placed in `BuildOutput\GVFS.Installer.Windows\bin\x64\<Debug or Release>`
+3. Run the tests **with elevation**.  Elevation is required because the functional tests create and delete a test service.
+
+   **Option 1:** Run the `GVFS.FunctionalTests.Windows` project from inside Visual Studio launched as Administrator.
+   
+   **Option 2:** Run `Scripts\RunFunctionalTests.bat` from CMD launched as Administrator.
+
+#### Extra and Full Suite
+
+By default, the functional tests run a subset of tests as a quick smoke test for developers.  There are three mutually exclusive arguments that can be passed to the functional tests to change this behavior:
+
+- `--full-suite`: Run all configurations of all functional tests
+- `--extra-only`: Run only those tests marked as "ExtraCoverage" (i.e. the tests that are not run by default)
+- `--windows-only`: Run only the tests marked as being Windows specific
+
+**NOTE** `Scripts\RunFunctionalTests.bat` already uses some of these arguments.  If you run the scripts using `RunFunctionalTests.bat` consider modifying the script to use these arguments rather than passing them as arguments to the script.

```

Comment:
This is a little confusing because you say `run the scripts` (what scripts? should this be "tests") `consider modifying the script`(what script?)

Reply:
Thanks you're right that this should be "tests"